### PR TITLE
MLS R package management bug fixing

### PR DESCRIPTION
**Pull request by llali**
This PR fixes couple issues in R package management in machine learning extension. 

- To get R available packages, the extension was running a query which doesn't work in all instance. So for now using an API to verify if the package name is valid. Also sqlmlutils doesn't support installing specific version for r so for now just installing the latest 
- The config path for the extension was wrong when installing the extension in ADS.

### CODE REVIEW

File: extensions/machine-learning-services/src/configurations/config.json [hunk 1]
```diff
@@ -7,5 +7,7 @@
 	"requiredRPackages": [
 		{ "name": "RODBCext", "repository": "https://cran.microsoft.com" },
 		{ "name": "sqlmlutils", "fileName": "sqlmlutils_0.7.1.zip", "downloadUrl": "https://github.com/microsoft/sqlmlutils/blob/master/R/dist/sqlmlutils_0.7.1.zip?raw=true"}
-	]
+	],
+
+	"rPackagesRepository": "http://cran.r-project.org"
```
Comment (Charles-Gagnon) [inline extensions/machine-learning-services/src/configurations/config.json]: This should be https

File: extensions/machine-learning-services/src/configurations/config.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export class Config {
 	 * Loads the config values
 	 */
 	public async load(): Promise<void> {
-		const rawConfig = await fs.readFile(path.join(this._root, 'src', 'configurations', configFileName));
+		const rawConfig = await fs.readFile(path.join(this._root, 'out', 'configurations', configFileName));
```
Comment (Charles-Gagnon) [inline extensions/machine-learning-services/src/configurations/config.ts]: Normally we put these at the root so you don't have to hardcode the paths like this

File: extensions/machine-learning-services/src/packageManagement/sqlRPackageManageProvider.ts [hunk 3]
```diff
@@ -93,20 +99,22 @@ export class SqlRPackageManageProvider extends SqlPackageManageProviderBase impl
...
+
 	/**
 	 * Returns package overview for given name
 	 * @param packageName Package Name
 	 */
 	protected async fetchPackage(packageName: string): Promise<nbExtensionApis.IPackageOverview> {
 		let packagePreview: nbExtensionApis.IPackageOverview = {
 			name: packageName,
-			versions: [],
+			versions: [constants.latestVersion],
```
Comment (Charles-Gagnon) [inline extensions/machine-learning-services/src/packageManagement/sqlRPackageManageProvider.ts]: You sure you want to always fetch the latest? That might break people if an update is released before we can test it. A better idea might be to just specify a known good version and have that be a config value - then you can always document the way for users to update that value manually if they really need the latest version.

Comment (Benjin) [reply extensions/machine-learning-services/src/packageManagement/sqlRPackageManageProvider.ts]: I like the idea of being able to specify your own version, but I think we should flip it around: default to latest stable (maybe within a major version? idk), then document the way for users to update that value to stick to a specific version.  Two reasons are 1) we should trust package stewards with their own quality, and 2) we don't want to get in the business of pushing an update (even a small one with just an incrementation in a config) each time a package updates.

File: extensions/machine-learning-services/src/test/packageManagement/sqlRPackageManageProvider.test.ts [hunk 4]
```diff
@@ -249,14 +249,14 @@ describe('SQL R Package Manager', () => {
...
 			'summary': ''
 		};
-		let allPackages = [{
-			'name': 'a-name',
-			'version': '1.1.2'
-		}];
-		testContext.queryRunner.setup(x => x.getRAvailablePackages(TypeMoq.It.isAny())).returns(() => Promise.resolve(allPackages));
+
+		testContext.httpClient.setup(x => x.fetch(TypeMoq.It.isAny())).returns(() => {
+			return Promise.resolve(``);
```
Comment (Benjin) [inline extensions/machine-learning-services/src/test/packageManagement/sqlRPackageManageProvider.test.ts]: any reason for backticks over normal single quotes?

File: extensions/machine-learning-services/src/configurations/config.ts (hunk 2 above)
Comment (llali) [reply extensions/machine-learning-services/src/configurations/config.ts]: yeah I put it inside the configuration folder. but will fix later in other PRs

---
In reply to: [369282200](https://github.com/microsoft/azuredatastudio/pull/8920#discussion_r369282200) [](ancestors = 369282200)

File: extensions/machine-learning-services/src/packageManagement/sqlRPackageManageProvider.ts (hunk 3 above)
Comment (llali) [reply extensions/machine-learning-services/src/packageManagement/sqlRPackageManageProvider.ts]: don't have any choice right now. the library I'm using doesn't get version

---
In reply to: [369283258](https://github.com/microsoft/azuredatastudio/pull/8920#discussion_r369283258) [](ancestors = 369283258)

**Pull request merged**